### PR TITLE
Gpu si pixel raw to digi vi1

### PR DESCRIPTION
**Pull request by VinInn**
mods by Cesare

### CODE REVIEW

**Comment by VinInn** [comment]
@calabria

File: EventFilter/SiPixelRawToDigi/plugins/SiPixelFedCablingMapGPU.cc [hunk 1]
```diff
@@ -25,8 +25,8 @@ void processCablingMap(SiPixelFedCablingMap const& cablingMap, SiPixelFedCabling
   std::vector<unsigned int>  RawId(MAX_SIZE);
   std::vector<unsigned int>  rocInDet(MAX_SIZE);
   std::vector<unsigned int>  moduleId(MAX_SIZE);
-  std::vector<unsigned char> badRocs(MAX_SIZE);
-  std::vector<unsigned char> modToUnp(MAX_SIZE);
+  std::vector<short int>     badRocs(MAX_SIZE);
+  std::vector<short int>     modToUnp(MAX_SIZE);
```
Comment (fwyzard) [inline EventFilter/SiPixelRawToDigi/plugins/SiPixelFedCablingMapGPU.cc]: why `short int` ?
these are always `bool` in the code

Comment (calabria) [reply EventFilter/SiPixelRawToDigi/plugins/SiPixelFedCablingMapGPU.cc]: I have already tried with bool but it did not work

**Comment by fwyzard** [comment]
Hi Cesare,
bool does not work because the storage for an std::vector<bool> is not the
same as an array of bool: a vector will pack multiple bools into a byte,
while the array does not.

I see that you made a change from char to short: did you see any problems
with char ?

Eventually, if the buffer size is not negligible, we could indeed look into
packing the bools; but we should measure the impact first.

.A

**Comment by calabria** [comment]
if I remember well using char the elements in the map were empty instead of being false or 0.

**Pull request merged**